drivers/i2c/i2c_ll_stm32.c

```diff
@@ -164,17 +157,15 @@ static int i2c_stm32_transfer(struct device *dev, struct i2c_msg *msg,
 		num_msgs--;
 	}
 exit:
-#if defined(CONFIG_I2C_STM32_V1)
-	LL_I2C_Disable(i2c);
-#endif
 	k_sem_give(&data->bus_mutex);
+	stm32_i2c_master_finish(dev);
```

Comment:
As master, could we use next_msg_flags pointer and disable I2C when there is no msg left to send?
For example: https://gist.github.com/ydamigos/8440e33d32da859aa566182ae3c6316b

Reply:
Good idea. Thank you for this example. I've combined "start/end" activity to use same functions for poll/interrupt modes.